compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -78,7 +78,7 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
                     Concrete,
                 }
                 let mut failure_kind = FailureKind::Concrete;
-                walk_abstract_const(tcx, ct, |node| match node {
+                walk_abstract_const::<_, ()>(tcx, ct, |node| match node {
```

Comment:
```suggestion
                walk_abstract_const::<!, _>(tcx, ct, |node| match node {
```
We probably also want to refactor the closure here but not as part of this PR